schema.cue

```diff
@@ -0,0 +1,166 @@
+// Line-protocol test corpus.
+package lpcorpus
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"crypto/md5"
+)
+
+// #DecodeInput specifies an input to the decoder.
+#DecodeInput: {
+	key: "\(hex.Encode(md5.Sum(json.Marshal({
+		"text":        text // Note: this has potential for ambiguity.
+		"defaultTime": defaultTime
+		"precision":   precision
+	}))))"
+	text: string | {
+		// binary holds base64-encoded text.
+		binary: string
+	}
+	// defaultTime holds the default time to use when
+	// there is no time in a line-protocol data point,
+	// in nanoseconds since the Unix epoch.
+	defaultTime: int
+	// precision holds the precision to use when interpreting
+	// timestamps.
+	precision:   #Precision
+
+	// implementation shows which implementation this
+	// test input came from.
+	implementation: string
+}
+
+// EncodeInput specifies an input to the encoder.
+#EncodeInput: {
+	key: "\(hex.Encode(md5.Sum(json.Marshal({
+		"point":             point
+		"omitInvalidFields": omitInvalidFields
+		"uintSupport":       uintSupport
+		"precision":         precision
+	}))))"
+	// point specifies the point to encode.
+	point:             #Point
+
+	// omitInvalid fields specifies whether invalid fields
+	// should be omitted from the encoded result.
+	// Not all encoders support this option.
+	omitInvalidFields: bool | *false
+
+	// uintSupport specifies whether unsigned integers
+	// are supported. All new encoders should support
+	// this option.
+	uintSupport:       bool | *true
+
+	// precision specifies the precision of the timestamps to
+	// include in the result.
+	precision:         #Precision
+
+	// implementation shows which implementation this
+	// test input came from.
+	implementation: string
+}
+
+#Precision: "1ns" | "1µs" | "1ms" | "1s" | "1m" | "1h"
+
+#Point: {
+	time: int
+	name: #StringOrBinary
+	tags: [... #Tag]
+	fields: [... #Field]
+}
+
+foo: #Point & {
```

Comment:
I probably don't understand this, but I found the `foo` here confusing

Reply:
It's a debugging remnant. You were right to be confused.